test/test_oneview_power_device.py

```diff
@@ -0,0 +1,431 @@
+                                        "data":
+                                            {"field": "invalid"}}
+        mock_ansible_module.return_value = mock_ansible_instance
+
+        PowerDeviceModule().run()
+
+        mock_ansible_instance.fail_json.assert_called_once_with(
+            msg=POWER_DEVICE_MANDATORY_FIELD_MISSING
+        )
+
+    @mock.patch.object(OneViewClient, 'from_json_file')
+    @mock.patch('oneview_power_device.AnsibleModule')
+    def test_should_not_add_when_add_raises_exception(self, mock_ansible_module, mock_ov_client_from_json_file):
+        mock_ov_instance = mock.Mock()
+        mock_ov_instance.power_devices.get_by.return_value = []
+        mock_ov_instance.power_devices.add.side_effect = Exception(FAKE_MSG_ERROR)
+
+        mock_ov_client_from_json_file.return_value = mock_ov_instance
+        mock_ansible_instance = create_ansible_mock(YAML_POWER_DEVICE)
+        mock_ansible_module.return_value = mock_ansible_instance
+
+        self.assertRaises(Exception, PowerDeviceModule().run())
+
+        mock_ansible_instance.fail_json.assert_called_once_with(
+            msg=FAKE_MSG_ERROR
+        )
+
+
+class PowerDeviceAbsentStateSpec(unittest.TestCase):
+    @mock.patch.object(OneViewClient, 'from_json_file')
+    @mock.patch('oneview_power_device.AnsibleModule')
+    def test_should_remove_power_device(self, mock_ansible_module, mock_ov_client_from_json_file):
+        mock_ov_instance = mock.Mock()
+        mock_ov_instance.power_devices.get_by.return_value = [{'name': 'name'}]
+
+        mock_ov_client_from_json_file.return_value = mock_ov_instance
+        mock_ansible_instance = create_ansible_mock(YAML_POWER_DEVICE_ABSENT)
+        mock_ansible_module.return_value = mock_ansible_instance
+
+        PowerDeviceModule().run()
+
+        mock_ansible_instance.exit_json.assert_called_once_with(
+            ansible_facts={},
+            changed=True,
+            msg=POWER_DEVICE_DELETED
+        )
+
+    @mock.patch.object(OneViewClient, 'from_json_file')
+    @mock.patch('oneview_power_device.AnsibleModule')
+    def test_should_do_nothing_when_power_device_not_exist(self, mock_ansible_module, mock_ov_client_from_json_file):
+        mock_ov_instance = mock.Mock()
+        mock_ov_instance.power_devices.get_by.return_value = []
+
+        mock_ov_client_from_json_file.return_value = mock_ov_instance
+        mock_ansible_instance = create_ansible_mock(YAML_POWER_DEVICE_ABSENT)
+        mock_ansible_module.return_value = mock_ansible_instance
+
+        PowerDeviceModule().run()
+
+        mock_ansible_instance.exit_json.assert_called_once_with(
+            ansible_facts={},
+            changed=False,
+            msg=POWER_DEVICE_ALREADY_ABSENT
+        )
+
+    @mock.patch.object(OneViewClient, 'from_json_file')
+    @mock.patch('oneview_power_device.AnsibleModule')
+    def test_should_not_remove_when_oneview_exception(self, mock_ansible_module, mock_ov_client_from_json_file):
```

Comment:
I suggest change the name of the method to `test_should_fail_when_remove_raises_exception`